lib/ansible/modules/remote_management/oneview/oneview_datacenter.py

```diff
@@ -0,0 +1,174 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: oneview_datacenter
+short_description: Manage OneView Datacenter resources
+description:
+    - "Provides an interface to manage Datacenter resources. Can add, update, and remove."
+version_added: "2.5"
+requirements:
+    - "hpOneView >= 3.1.0"
+author:
+    - Alex Monteiro (@aalexmonteiro)
+    - Madhav Bharadwaj (@madhav-bharadwaj)
+    - Priyanka Sood (@soodpr)
+    - Ricardo Galeno (@ricardogpsf)
+options:
+    state:
+        description:
+            - Indicates the desired state for the Datacenter resource.
+              C(absent) will remove the resource from OneView, if it exists.
+              C(present) will ensure the datacenter resource exists and its properties are compliant with HPE OneView.
+        choices: ['present', 'absent']
+        default: present
+    data:
+        description:
+            - List with Datacenter properties and its associated states.
+        required: true
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.validateetag
+'''
+
+EXAMPLES = '''
+- name: Add a Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        contents:
+            # You can choose either resourceName or resourceUri to inform the Rack
+            - resourceName: '{{ datacenter_content_rack_name }}' # option 1
+              resourceUri: ''                                    # option 2
+              x: 1000
+              y: 1000
+        depth: 6000
+        name: "MyDatacenter"
+        width: 5000
+  no_log: true
+  delegate_to: localhost
+
+- name: Update the Datacenter with specified properties (no racks)
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        contents: ~
+        coolingCapacity: '5'
+        coolingMultiplier: '1.5'
+        costPerKilowattHour: '0.10'
+        currency: USD
+        defaultPowerLineVoltage: '220'
+        deratingPercentage: '20.0'
+        depth: 5000
+        deratingType: NaJp
+        name: "MyDatacenter"
+        width: 4000
+  no_log: true
+  delegate_to: localhost
+
+- name: Rename the Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        name: "MyDatacenter"
+        newName: "My Datacenter"
+  no_log: true
+  delegate_to: localhost
+
+- name: Remove the Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: absent
+    data:
+        name: 'My Datacenter'
+  no_log: true
```

Comment:
As on other PR, update https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/oneview.py#L210 and remove `no_log: true` from examples